### PR TITLE
HCX Plug

### LINKED ISSUE #37: Seperate HCX into a plug

### LINKED ISSUE #6381: Replace useDispatch w. useQuery/request: HCX (src/Components/HCX/**)
**Is your feature request related to a problem? Please describe.**
Part of #6316

### LINKED ISSUE #8413: Improve mobile responsiveness and i18n for HCX components
Currently, HCX components are partially responsive, they break on small screens, get that fixed and along the way, also add i18n.

### PR DESCRIPTION
**Pull request by khavinshankar**
## Proposed Changes

- Fixes https://github.com/ohcnetwork/docs/issues/37
- Fixes https://github.com/ohcnetwork/care_fe/issues/8413
- Closes https://github.com/ohcnetwork/care_fe/issues/6381
- Depends on: https://github.com/ohcnetwork/care/pull/2399

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
@khavinshankar can you get this PR ready for review and testing

File: src/Common/hooks/useAsyncOptions.ts
Comment (rithviknishad) [inline src/Common/hooks/useAsyncOptions.ts:1]: `useAsyncOptions` is no longer needed with `useQuery` and `mergeQueryOptions` utility fn as mentioned in the doc. comment of the `useAsyncOptions`.

See how medibase autocomplete form field was migrated out of useAsyncOptions to use useQuery + mergeQueryOptions.

https://github.com/ohcnetwork/care_fe/pull/7381/files#diff-27f7661cf02d3f78f7bbc53331f501ddc59ad9a33c613022593b529c128c8bb4

Let's delete this file once migrated too.

File: src/Components/Common/PMJAYProcedurePackageAutocomplete.tsx [hunk 1]
```diff
@@ -41,7 +43,20 @@ export default function PMJAYProcedurePackageAutocomplete(props: Props) {
...
+              { query: { query, limit: 10 } },
+            );
+
+            if (res?.ok && data) {
+              return data;
+            }
+
+            return [];
+          })
+        }
```
Comment (rithviknishad) [inline src/Components/Common/PMJAYProcedurePackageAutocomplete.tsx]: Switch to `useQuery` + `mergeQueryOptions` instead.

File: src/Components/Facility/ConsultationClaims.tsx [hunk 2]
```diff
@@ -19,48 +20,46 @@ export default function ConsultationClaims({
...
+        if (res.data && res.data.results) {
+          if (isCreateLoading)
+            Notification.Success({ msg: "Fetched Claim Approval Results" });
+        } else {
+          if (isCreateLoading)
+            Notification.Error({
+              msg: "Error Fetching Claim Approval Results",
+            });
+        }
+      },
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationClaims.tsx]: i18n + reducing if else ladder for readability.

```suggestion
        if (!isCreateLoading) {
          return;
        }
        
        if (res.data && res.data.results) {
          Notification.Success({ msg: "Fetched Claim Approval Results" });
          return;
        }
        
        Notification.Error({ msg: "Error Fetching Claim Approval Results" });
      },
```

File: src/Components/Facility/DischargeModal.tsx [hunk 3]
```diff
@@ -81,6 +82,28 @@ const DischargeModal = ({
...
+      if (res?.res?.ok && res?.data?.results?.length !== 0) {
+        setLatestClaim(res?.data?.results[0]);
+        if (isCreateClaimLoading)
+          Notification.Success({ msg: "Fetched Claim Approval Results" });
+      } else {
+        setLatestClaim(undefined);
+        if (isCreateClaimLoading)
+          Notification.Success({ msg: "Error Fetched Claim Approval Results" });
+      }
+      setIsCreateClaimLoading(false);
```
Comment (rithviknishad) [inline src/Components/Facility/DischargeModal.tsx]: same, i18n + reduce if-else ladder.

File: src/Components/HCX/ClaimCardInfo.tsx [hunk 4]
```diff
@@ -38,39 +42,45 @@ export default function ClaimDetailCard({ claim }: IProps) {
           <span
             className={classNames(
               "rounded p-1 px-2 text-sm font-bold text-white shadow",
-              status === "Approved" && "bg-primary-400",
-              status === "Rejected" && "bg-danger-400",
-              status === "Pending" && "bg-yellow-400",
+              status === t("claim__status__approved") && "bg-primary-400",
+              status === t("claim__status__rejected") && "bg-danger-400",
+              status === t("claim__status__pending") && "bg-yellow-400",
```
Comment (rithviknishad) [inline src/Components/HCX/ClaimCardInfo.tsx]: This seems hacky. How about we store `status` in known set of options in English and translate only in the place where we render it?

File: cypress.config.ts [hunk 5]
```diff
@@ -34,5 +34,6 @@ export default defineConfig({
   },
   env: {
     API_URL: process.env.REACT_CARE_API_URL ?? "http://localhost:9000",
+    ENABLE_HCX: process.env.REACT_ENABLE_HCX ?? false,
```
Comment (rithviknishad) [inline cypress.config.ts:37]: Should we make plugs configurable in cypress? Shouldn't we enable all plugs for cypress tests? (Also by ensuring all plugs are enabled in FE build too for cypress?)

File: src/Components/HCX/InsurerAutocomplete.tsx [hunk 6]
```diff
@@ -35,7 +37,20 @@ export default function InsurerAutocomplete(props: Props) {
...
+              { query: { query, limit: 10 } },
+            );
+
+            if (res?.ok && data) {
+              return data;
+            }
+
+            return [];
+          })
+        }
```
Comment (rithviknishad) [inline src/Components/HCX/InsurerAutocomplete.tsx]: Switch to `useQuery` + `mergeQueryOptions` instead.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: cypress.config.ts (hunk 5 above)
Comment (khavinshankar) [reply cypress.config.ts:37]: Yes, that's a nice idea, but it is a bit tricky for HCX as we have 2 flows based on whether HCX is enabled. Creating a new issue to track this https://github.com/ohcnetwork/care_fe/issues/8553

**Review by rithviknishad — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Pull request merged**